Enable using a shell other than bash.
Courtesy "Martin Guy" <martinwguy@yahoo.it>
http://sourceware.org/ml/crossgcc/2008-08/msg00037.html

--- binutils-2.18.orig/ld/genscripts.sh	2007-08-12 20:00:07.000000000 +0100
+++ binutils-2.18/ld/genscripts.sh	2008-08-25 16:09:54.000000000 +0100
@@ -398,9 +398,11 @@
     . $em_script
     em_script=$current_script
   }
-  fragment()
+  
+  alias fragment='em_lineno=$LINENO; fragment_em'
+  fragment_em()
   {
-    local lineno=$[${BASH_LINENO[0]} + 1]
+    local lineno=$[$em_lineno + 1]
     echo >> e${EMULATION_NAME}.c "#line $lineno \"$em_script\""
     cat >> e${EMULATION_NAME}.c
   }
